Feature/819 attains download links

maxdiebold-erg opened the pull request:
## Related Issues:
* [HMW-819](https://jira.epa.gov/browse/HMW-819)

## Main Changes:
* Added links to the Expert Query API & dashboard throughout the application. The former is used for direct downloads, the latter for _Advanced Filtering_. These links are placed in the following locations:
  * Community page's **Overview** tab
    * All waterbodies in watershed: direct data download
    * Each waterbody in the list: direct data download & link to portal
  * Community page's **Identified Issues** tab (**Impaired Assessed Waters** section)
    * Download data for all waterbodies in the watershed with the selected impairments
  * Community page's **Restore** tab (**Restoration Plans** section)
    * All plans: direct data download
    * Each plan: direct data download & link to portal
  * **Plan Summary** page
    * direct data download & link to portal

## Steps To Test:
1. For each of the locations listed above, test the download links and confirm that the proper parameters are being passed. Confirm the result set is showing appropriate results.
2. If there are filter controls, adjust the controls and repeat step 1.
3. If there is an _Advanced Filtering_ link, verify it links the Expert Query ATTAINS dashboard with the proper inputs populated.

cschwinderg reviewed (changes requested):
Looks good and works well!

I think there are a couple more places this could be added. 
* Restore/Protect tab popups
![image](https://github.com/user-attachments/assets/9158035a-1f84-44ce-9e01-232439ec29f8)

* Identified Issues tab popup
![image](https://github.com/user-attachments/assets/87c91e86-7d68-4f02-bbab-372275d6576a)

* Protect tab => Protection Projects accordion (should work the same as the restore tab). 040302020902 is a good huc for this one.
![image](https://github.com/user-attachments/assets/af72d79b-ae05-4207-8381-85da975ca6b7)

* Can you also add a couple of tests for this as well?

maxdiebold-erg commented:
@cschwinderg I've added the links in the places you mentioned and added a couple tests. I can work on more tests but those should cover most of the component code.

I didn't add the top-level download links to the **Protection Projects** list of the Protect tab, because that list can contain both GRTS and ATTAINS projects and I thought it would be misleading. I just added the links to individual ATTAINS projects in the list.

Also, I added the `overallStatus: 'Not Supporting'` filter to the downloads on the Identified Issues tab (both in the list and the popups).

cschwinderg reviewed (approved):
Thanks for making those changes and adding the tests.

I'm ok with these two tests for now. I'm mainly trying to keep our coverage above 80%. We can add more if we fall below 80. We should try to add tests whenever we add new code, when possible. I've never been very good about doing that. Now it's a contract requirement, so I'm going to try to do a better job of keeping up with tests. 

I also pushed up an additional commit to fix an issue (screenshot below) with switching to the ArcGIS Online attains layer, as opposed to the gispub version. Basically, the new AGO services may not return the attributes immediately, which causes the geometry type to default to point. I just added a line to check the actual geometry first and then fall back to the old way if there is no geometry.

![image](https://github.com/user-attachments/assets/c8c14172-fe44-4614-b240-860feecfc7a3)
